uimaj-core/src/main/java/org/apache/uima/cas/impl/BinaryCasSerDes.java

```diff
@@ -1697,12 +1700,23 @@ private void createFSsFromHeaps(boolean isDelta, int startPos, CommonSerDesSeque
                // start of normal non-array
         CASImpl view = null; 
         boolean isSofa = false;
+        boolean documentAnnotationPreviouslyIndexed = false;
         if (type.isAnnotationBaseType()) {
           Sofa sofa = getSofaFromAnnotBase(heapIndex, stringHeap, addr2fs, csds);  // creates sofa if needed and exists (forward ref case)
           view = (sofa == null) ? baseCas.getInitialView() : baseCas.getView(sofa);
           if (type == tsi.docType) {
-            fs = view.getDocumentAnnotation();  // creates the document annotation if it doesn't exist
-            view.removeFromCorruptableIndexAnyView(fs, view.getAddbackSingle());
+            Annotation documentAnnotationPrevious = view.getDocumentAnnotationNoCreate();
+            if (documentAnnotationPrevious == null) {
+              // document annotation not present
+              fs = view.createDocumentAnnotationNoRemoveNoIndex(0); // create but don't index
+              view.set_deserialized_doc_annot_not_indexed((Annotation)fs);  // for use by other code that sets length, if this is not indexed

```

Comment:
length of what?

Reply:
The Document Annotation is a subtype of "Annotation" and has a begin and end.  The begin is normally defaulted to 0.  When the sofa data is a "string", the "end" is updated with the length of the sofa data string.  (although I cannot find this in the docs, it's what the framework has been doing forever...)